lib/galaxy/webapps/galaxy/api/users.py

```diff
@@ -327,6 +329,45 @@ def index(
     ) -> List[UserModel]:
         return self.service.get_index(trans=trans, deleted=deleted, f_email=f_email, f_name=f_name, f_any=f_any)
 
+    @router.post(
+        "/api/users",
+        name="Create user",
+        summary="Creates a new Galaxy user.",
+    )
+    def create(
+        self,
+        trans: ProvidesUserContext = DependsOnTrans,
+        payload: CreateUserPayload = CreateUserBody,
+    ) -> CreatedUserModel:
+        """
+        Rework CreateUserPayload payload and add pydantic model for return
+        """
+        if not trans.app.config.allow_user_creation and not trans.user_is_admin:
+            raise exceptions.ConfigDoesNotAllowException("User creation is not allowed in this Galaxy instance")
+
+        # TODO: this is not working, because we cannot access get_or_create_remote_user
```

Comment:
I am unable to migrate this 'get_or_create_remote_user' to 'UsersService' because it uses `self.environ` and i do not know how to access it. Could anybody help me with this?